### PR TITLE
ART-6673 notify QE release ready through update QE release repo

### PR DESCRIPTION
**Pull request by Ximinhan**
Ref [ART-6673](https://issues.redhat.com/browse/ART-6673), talked with QE to create a pr to QE [release-tests repo](https://github.com/openshift/release-tests), commit the release info in their repo. QE then will know the release is ready for test

the created pr looks like https://github.com/openshift/release-tests/pull/188

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/promote.py [hunk 1]
```diff
@@ -1519,13 +1523,52 @@ def handle_qe_notification(self, release_jira: str, release_name: str, impetus_a
...
+            self._logger.warning("release file not found in upstream repo, skip update qe repo")
+            return
+        # update release file
+        file_content_bytes = yaml.dump(file).encode()
+        fork_file = fork_repo.get_contents(file_path, ref=release_name)
+        fork_repo.update_file(file_path, update_message, file_content_bytes, fork_file.sha, branch=release_name)
+        # create pr
+        pr = upstream_repo.create_pull(title=update_message, body=update_message, base="z-stream", head=f"openshift-bot:{release_name}")
+        pr.add_to_labels("lgtm", "approved")
+        pr.merge()
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:1569]: Can we wrap this and other interactions with GitHub in `try`? Let's not have the promote job fail if this fails.

File: pyartcd/pyartcd/pipelines/promote.py [hunk 2]
```diff
@@ -1519,13 +1523,52 @@ def handle_qe_notification(self, release_jira: str, release_name: str, impetus_a
...
+        # create branch
+        for branch in fork_repo.get_branches():
+            if branch.name == release_name:
+                fork_repo.get_git_ref(f"heads/{release_name}").delete()
+        fork_branch = fork_repo.create_git_ref(f"refs/heads/{release_name}", upstream_repo.get_branch("z-stream").commit.sha)
+        self._logger.info("Created fork branch ref %s", fork_branch.ref)
+        # get release file
+        try:
+            release_content = upstream_repo.get_contents(file_path, ref="z-stream")
+            file = yaml.load(release_content.decoded_content, Loader=yaml.FullLoader)
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:1553]: Misleading variable name. The result of `yaml.load()` is not a file, but a data structure.

File: pyartcd/pyartcd/pipelines/promote.py [hunk 3]
```diff
@@ -39,6 +39,9 @@
 from pyartcd.oc import get_release_image_info, get_release_image_pullspec, extract_release_binary, \
     extract_release_client_tools, get_release_image_info_from_pullspec, extract_baremetal_installer
 from pyartcd.runtime import Runtime, GroupRuntime
+from github import Github
+from github import GithubException
+from yaml.parser import ParserError
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:44]: Think the ruamel library has an exception that can be caught, let's not introduce the stdlib yaml parser here.

File: pyartcd/pyartcd/pipelines/promote.py (hunk 1 above)
Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/promote.py]: updated in https://github.com/openshift-eng/art-tools/pull/672/commits/995384aa85e26ab602932b4d690eec6b3f944a3c

File: pyartcd/pyartcd/pipelines/promote.py (hunk 2 above)
Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/promote.py]: updated in https://github.com/openshift-eng/art-tools/pull/672/commits/995384aa85e26ab602932b4d690eec6b3f944a3c

File: pyartcd/pyartcd/pipelines/promote.py (hunk 3 above)
Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/promote.py]: updated in https://github.com/openshift-eng/art-tools/pull/672/commits/995384aa85e26ab602932b4d690eec6b3f944a3c

**Pull request merged**